server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTable.java

```diff
@@ -0,0 +1,303 @@
+      .collect(toList());
+  }
+
+  private class PropertyRow {
+
+    private final Long id;
+    private final String key;
+    private final Long resourceId;
+    private final String value;
+    private final Long createdAt;
+
+    public PropertyRow(long id, String key, Long resourceId, String value, Long createdAt) {
+      this.id = id;
+      this.key = key;
+      this.resourceId = resourceId;
+      this.value = value;
+      this.createdAt = createdAt;
+    }
+
+    public Long id() {
+      return id;
+    }
+
+    public String key() {
+      return key;
+    }
+
+    public Long resourceId() {
+      return resourceId;
+    }
+
+    public String value() {
+      return value;
+    }
+
+    public Long createdAt() {
+      return createdAt;
+    }
+
+    @Override
+    public String toString() {
+      return "{" +
+        "id=" + id +
+        ", key='" + key + '\'' +
+        ", resourceId=" + resourceId +
+        ", value='" + value + '\'' +
+        ", createdAt=" + createdAt +
+        '}';
+    }
+  }
+
+  private class Webhook {
+
+    private final PropertyRow name;
+    private final PropertyRow url;
+    private String organisationUuid;
+    private String projectUuid;
+
+    public Webhook(@Nullable PropertyRow name, @Nullable PropertyRow url, @Nullable String organisationUuid, @Nullable Map<Long, String> projectsKeys) {
+      this.name = name;
+      this.url = url;
+      this.organisationUuid = organisationUuid;
+      if (StringUtils.isBlank(organisationUuid)) {
+        this.projectUuid = projectsKeys.get(name.resourceId());
+      }
+    }
+
+    public String name() {
+      return name.value();
+    }
+
+    public String url() {
+      return url.value();
+    }
+
+    public String organisationUuid() {
+      return organisationUuid;
+    }
+
+    public String projectUuid() {
+      return projectUuid;
+    }
+
+    public Long createdAt() {
+      return name.createdAt();
+    }
+
+    public List<Long> rowIds() {
+      List<Long> toBeDropped = new ArrayList<>();
+      if (name != null) {
+        toBeDropped.add(name.id());
+      }
+      if (url != null) {
+        toBeDropped.add(url.id());
+      }
+      return toBeDropped;
+    }
+
+    public boolean isValid() {
+      return name != null && url != null && name() != null && url() != null && (organisationUuid() != null || projectUuid() != null) && createdAt() != null;
+    }
+
+    @Override
+    public String toString() {
+      final StringBuilder s = new StringBuilder().append("Webhook{").append("name=").append(name);
+      if (name != null) {
+        s.append(name.toString());
+      }
+      s.append(", url=").append(url);
+      if (url != null) {
+        s.append(url.toString());
+      }
+      s.append(", organisationUuid='").append(organisationUuid).append('\'')
+        .append(", projectUuid='").append(projectUuid).append('\'').append('}');
+      return s.toString();
+    }
+  }
+
+  private class projectFlyweight {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Rename this class name to match the regular expression '^[A-Z][a-zA-Z0-9]*$'. ([squid:S00101](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00101))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-647%2FSONAR-10345_manage_webhooks&issues=AWGPwo1FFMFpNJFuf2rl&open=AWGPwo1FFMFpNJFuf2rl)

[ISSUE_KEY]: AWGPwo1FFMFpNJFuf2rl